physics/forkable_body.hpp

```diff
@@ -340,7 +340,7 @@ void Forkable<Tr4jectory>::WriteSubTreeToMessage(
   for (auto const& pair : children_) {
     Instant const& fork_time = pair.first;
     Tr4jectory const& child = pair.second;
-    if (!last_instant || fork_time != last_instant) {
+    if (!last_instant || fork_time != *last_instant) {
```

Comment:
I'm a bit puzzled by this change, as it seems to me that http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/n4480.html#optional.synop implies that this should be legal without the `*`.  Are you sure that you are using the latest greatest version of `libwhateverc++`?  I am asking because your latest error log has many weird diagnostics related to `optional`.